src/test/java/fr/paris/lutece/portal/service/daemon/DaemonSchedulerTest.java

```diff
@@ -469,36 +475,35 @@ private void testUnScheduleWhileRunning( boolean shouldThrow ) throws ClassNotFo
             assertTrue( testDaemon.hasRun( ) );
             Thread.sleep( 10L ); // leave some time to the daemon to exit executing daemons
             assertEquals(1, testDaemon.getStopCallNumber( ) );
-            try
-            {
-                testDaemon.go( 1100L, TimeUnit.MILLISECONDS );
-                fail( "Daemon executed after unscheduling" );
-            }
-            catch ( TimeoutException e )
-            {
-                // OK
-            }
+
+            testDaemon.go( 1100L, TimeUnit.MILLISECONDS );
+            fail( "Daemon executed after unscheduling" );
+
+        }
+        catch( TimeoutException ee ) {
+            //ok
+        }
+        catch( InterruptedException | BrokenBarrierException  e)
+        {
+            fail("failed" + e + ":" + e.getMessage( ) );
```

Comment:
here you swallow the exception and you loose the stack trace which might be helpful. Why not let tests throw exceptions, which makes them fail ?